core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeHttpRequestStage.java

```diff
@@ -59,17 +59,20 @@ public MakeHttpRequestStage(HttpClientDependencies dependencies) {
     }
 
     private HttpExecuteResponse executeHttpRequest(SdkHttpFullRequest request, RequestExecutionContext context) throws Exception {
+        MetricCollector metricCollector = context.metricCollector();
```

Comment:
Not a comment, just a small question. What's the difference between these two collectors?

Reply:
`metricCollector` here is the collector for all API call attempt metrics, the name is `ApiCallAttempt`. The code is creating a child collector of this collector, `HttpClient` and passing that as a collector for the HTTP client so that any metrics the HTTP client reports are added to this collector instead so it's clear from the node path later, `ApiCallAttempt > HttpClient` that the metrics were reported by the HTTP client.

Reply:
It's clear now. Thanks.